speedy-recovery-client/src/components/secured/home/HomePageFactory.js

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from "react";
+import HomePageForPatient from "../home/HomePageForPatient";
+import HomePageForParent from "../home/HomePageForParent";
+import HomePageForPractitioner from "../home/HomePageForPractitioner";
+
+class HomePageFactory extends Component {
+  render() {
+    const { user, patient, events } = this.props;
+
+    switch (user.role) {
+      case "Patient":
+        return (
+          <HomePageForPatient
+            user={user}
+            event={
+              events
+                .filter(
+                  event =>
+                    new Date(event.start) - new Date() > 0
+                )
+                .sort(function(a, b) {
+                  return new Date(a.start) - new Date(b.start);
+                })[0]
+            }
+          />
+        );
+      case "Parent":
+        return (
+          <HomePageForParent
+            user={user}
+            event={
+              events
+                .filter(
+                  event =>
+                    new Date(event.start) - new Date() > 0
+                )
+                .sort(function(a, b) {
+                  return new Date(a.start) - new Date(b.start);
+                })[0]
+            }
+            patient={patient}
+          />
+        );
+      case "Practitioner":
+        return (
+          <HomePageForPractitioner
+            user={user}
+            event={
```

Comment:
The event filtering and mapping is the same every time. Do it outside the conditional, so you just have to do it once